### ISSUE TITLE
luacheck does not warn about accessing nonexistent members of table

### ISSUE THREAD

**Issue by crabdancing** [mpeterv/luacheck#205]
I create code with the following contents:

```lua
local some_namespace = {}
some_namespace.nonexistent_function()
print(some_namespace.whyyyyy)
```

Result:

```
$ luacheck badcode.lua
Checking badcode.lua                              OK

Total: 0 warnings / 0 errors in 1 file
```

**Comment by Seadragon91** [comment]
Please read #198

**Comment by crabdancing** [comment]
Aww that's really sad :c I hope you find someone who can help take care of this repo.

**Comment by alerque** [comment]
I don't think this is something `luacheck` is ever going to be able to check. You would have to *execute* the code in order to know for sure what was or wasn't in a Lua table. There is no way to know with any confidence from just lexing the language as you show. Also Lua has ways of handling this where calling non-existing methods can actually create/load them etc.

If you still feel this is something that needs more feedback please open a new issue on the [new official repository here](https://github.com/lunarmodules/luacheck). In the mean time please consider closing this issue to help tidy up as there is nobody with permissions to do so—only issue authors can close their own.
